sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java

```diff
@@ -1471,25 +1443,79 @@ ChatCompletions getChatCompletionsWithAzureExtensions(String deploymentOrModelNa
     }
 
     /**
-     * Starts the generation of a batch of images from a text caption.
+     * Creates an image given a prompt.
+     * <p>
+     * <strong>Request Body Schema</strong>
+     * </p>
+     * <pre>{@code
+     * {
+     *     model: String (Optional)
+     *     prompt: String (Required)
+     *     n: Integer (Optional)
+     *     size: String(256x256/512x512/1024x1024/1792x1024/1024x1792) (Optional)
+     *     response_format: String(url/b64_json) (Optional)
+     *     quality: String(standard/hd) (Optional)
+     *     style: String(natural/vivid) (Optional)
+     *     user: String (Optional)
+     * }
+     * }</pre>
+     * <p>
+     * <strong>Response Body Schema</strong>
+     * </p>
+     * <pre>{@code
+     * {
+     *     created: long (Required)
+     *     data (Required): [
+     *          (Required){
+     *             url: String (Optional)
+     *             b64_json: String (Optional)
+     *             revised_prompt: String (Optional)
+     *         }
+     *     ]
+     * }
+     * }</pre>
+     *
+     * @param deploymentOrModelName Specifies either the model deployment name (when using Azure OpenAI) or model name
+     * (when using non-Azure OpenAI) to use for this request.
+     * @param imageGenerationOptions Represents the request data used to generate images.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return the result of a successful image generation operation along with {@link Response}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<BinaryData> getImageGenerationsWithResponse(String deploymentOrModelName,
+        BinaryData imageGenerationOptions, RequestOptions requestOptions) {
```

Comment:
Feel that `BinaryData` is overused. We should have a method 
`public Response<ImageGenerations> getImageGenerationsWithResponse(String deploymentOrModelName,
        imageGenerationOptions imageGenerationOptions, RequestOptions requestOptions)`

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/36859/files#diff-b477cd2968cae4b347314275298a5f5c2f49926f547869eba6060b0874adb43dR10

Reply:
This is a good reminder that we should update the release notes too!